frontend/src/lib/components/Cards/TextCard/TextCard.tsx

```diff
@@ -0,0 +1,289 @@
+                                    value={value}
+                                    onChange={(newValue) => onChange(newValue)}
+                                />
+                                <div className="text-muted inline-flex items-center space-x-1">
+                                    <IconMarkdown />
+                                    <span>Basic formatting and markdown support</span>
+                                </div>
+                            </Tabs.TabPane>
+                            <Tabs.TabPane tab="Preview" key={'preview-card'}>
+                                <TextCardBody text={value} />
+                            </Tabs.TabPane>
+                        </Tabs>
+                    )}
+                </Field>
+            </Form>
+        </LemonModal>
+    )
+}
+
+interface TextCardProps extends React.HTMLAttributes<HTMLDivElement>, Resizeable {
+    dashboardId?: string | number
+    textTile: DashboardTile
+    children?: JSX.Element
+    updateColor?: (newColor: Tileable['color']) => void
+    removeFromDashboard?: () => void
+    duplicate?: () => void
+    moveToDashboard?: (dashboard: DashboardType) => void
+    /** buttons to add to the "more" menu on the card**/
+    moreButtons?: JSX.Element | null
+}
+
+function TextCardBody({ text }: { text: string }): JSX.Element {
+    return (
+        <div className={'w-full h-full block'}>
+            <Textfit mode={text?.match(/([\r\n])/gm)?.length ? 'multi' : 'single'} min={16} max={100}>
+                <div className="TextCard-Body px-2 pb-2">
+                    <ReactMarkdown>{text}</ReactMarkdown>
+                </div>
+            </Textfit>
+        </div>
+    )
+}
+
+export function TextCardInternal(
+    {
+        textTile,
+        showResizeHandles,
+        canResizeWidth,
+        children,
+        className,
+        dashboardId,
+        updateColor,
+        moreButtons,
+        removeFromDashboard,
+        duplicate,
+        moveToDashboard,
+        ...divProps
+    }: TextCardProps,
+    ref: React.Ref<HTMLDivElement>
+): JSX.Element {
+    const { push } = useActions(router)
+    const { text } = textTile
+    if (!text) {
+        throw new Error('TextCard requires text')
+    }
+
+    const { nameSortedDashboards } = useValues(dashboardsModel)
+    const otherDashboards = nameSortedDashboards.filter((dashboard) => dashboard.id !== dashboardId)
+    return (
+        <div
+            className={clsx('TextCard border rounded flex flex-col overflow-hidden', className)}
+            data-attr="text-card"
+            {...divProps}
+            ref={ref}
+        >
+            <div className={'flex flex-row px-2 pt-2'}>
+                {textTile.color &&
+                    textTile.color !==
+                        InsightColor.White /* White has historically meant no color synonymously to null */ && (
+                        <div className={clsx('InsightMeta__ribbon ml-2', textTile.color)} />
+                    )}
```

Comment:
Honestly the ribbon is not particularly useful on insight cards, but on these text cards it's really useless. I propose dropping it from scope for now. We can later figure _how_ should text cards colors work if we prioritize that.
<img width="312" alt="Screen Shot 2022-10-11 at 17 40 02" src="https://user-images.githubusercontent.com/4550621/195137157-558106bf-9bec-4940-8785-7533071e530a.png">